src/poliastro/threebody/restricted.py

```diff
@@ -0,0 +1,173 @@
+# coding: utf-8
+
+import numpy as np
+from scipy.optimize import root
+
+
+def lagrange_points(r12, m1, m2):
+    """Computes the Lagrangian points using the formulation
+    in Problem 8-5 of Battin, Richard H. 'An introduction to 
+    the mathematics and methods of astrodynamics'. AIAA, 1999.
+
+    Parameters
+    ----------
+    r12 : float
+        Collinear distance
+    m1 : float
+        Mass of the main body
+    m2 : float
+        Mass of the secondary body
+
+    Returns
+    -------
+    array(float)
+        Distance of the Lagrangian points to the center,
+        projected on the collinear line
+    """
+
+    rho = r12
+
+    def eq_L1(rho2):
+        aux = 3 * rho**2 * rho2 - 3 * rho * rho2**2 + rho2**3
+        aux *= rho2**2
+        aux /= rho**3 - rho2**3
+        aux /= (rho - rho2)**2
+        aux -= m2 / m1
+        return aux
+
+    def eq_L2(rho2):
+        aux = 3 * rho**2 * rho2 + 3 * rho * rho2**2 + rho2**3
+        aux *= rho2**2
+        aux /= rho**3 - rho2**3
+        aux /= (rho + rho2)**2
+        aux -= m2 / m1
+        return aux
+
+    def eq_L3(rho1):
+        aux = 3 * rho**2 * rho1 + 3 * rho * rho1**2 + rho1**3
+        aux *= rho1**2
+        aux /= rho**3 - rho1**3
+        aux /= (rho + rho1)**2
+        aux -= m1 / m2
+        return aux
+
+    l = np.zeros((5,))
+
+    # L1
+    rho2 = root(eq_L1, 0).x
+    rho1 = rho - rho2
+    l[0] = rho1
+
+    # L2
+    rho2 = root(eq_L2, 0).x
+    rho1 = rho + rho2
+    l[1] = rho1
+
+    # L3
+    rho1 = root(eq_L3, -1000).x
+    # rho2 = rho + rho1
+    l[2] = - rho1
+
+    # TODO: add checks to the results given by `root`
+    # TODO: reassure that the initial values given to `root` are good for all cases
+
+    l[3] = l[4] = 0.5 * r12
+
+    return l
+
+
+def lagrange_points_vec(m1, r1_, m2, r2_, n_):
+    """Computes the five Lagrange points in the Restricted Circular 3 Body
+    Problem. Returns the radiovectors in the same vectorial base as r1 and 
+    r2 for the five Lagrangian points: L1, L2, L3, L4, L5
+
+    Parameters
+    ----------
+    m1 : float
+        Mass of the main body. This body is the one with the biggest mass
+    r1_ : array
+        Radiovector of the main body
+    m2 : float
+        Mass of the secondary body
+    r2_: array
+        Radiovector of the secondary body
+    n_ : array
+        Normal vector to the plane in which the two orbits of the main
+        and the secondary body are contained
+
+    Returns
+    -------
+    LP: list of all Lagrange points
+    """
+
+    r1 = np.asarray(r1_).reshape((3,))
+    r2 = np.asarray(r2_).reshape((3,))
+    n = np.asarray(n_).reshape((3,))
+
+    # Define local reference system:
+    # Center: main body
+    # x axis: points to the secondary body
+    ux = r2 - r1
+    r12 = np.linalg.norm(ux)
```

Comment:
If we make `ux` a vector with units, we can compute the norm using `poliastro.util.norm`.